stream_alert/alert_processor/outputs/credentials/provider.py

```diff
@@ -0,0 +1,654 @@
+        if not credentials.is_encrypted():
+            LOGGER.error('Error: Writing unencrypted credentials to disk is disallowed.')
+            return False
+
+        raw_creds = credentials.data()
+
+        spool = tempfile.SpooledTemporaryFile()
+        spool.write(raw_creds)
+
+        key = self.get_spool_cache_key(descriptor)
+        SpooledTempfileDriver.SERVICE_SPOOLS[key] = spool
+
+        return True
+
+    @staticmethod
+    def clear():
+        """Clears all global spools.
+
+        De-allocating the spools triggers garbage collection, which implicitly closes the
+        file handles.
+        """
+        SpooledTempfileDriver.SERVICE_SPOOLS = {}
+
+    def offer_fileobj(self, descriptor):
+        """Opens a file-like temporary file spool and returns it.
+
+        If you use the return value in a `with` statement block then the file descriptor
+        auto-close.
+
+        NOTE: (!) This returns an ephemeral spool that is not attached to the caching mechanism
+            in save_credentials() and load_credentials()
+
+        Returns:
+            file object
+        """
+        return tempfile.SpooledTemporaryFile(0, 'a+b')
+
+    def get_spool_cache_key(self, descriptor):
+        return '{}/{}'.format(self._service_name, descriptor)
+
+
+class EphemeralUnencryptedDriver(CredentialsProvidingDriver, CredentialsCachingDriver):
+    """Stores credentials UNENCRYPTED on the Python runtime stack.
+
+    It is ephemeral and is only readable by the current Python process... hopefully.
+    """
+
+    CREDENTIALS_STORE = {}
+
+    def __init__(self, service_name):
+        self._service_name = service_name
+
+    def has_credentials(self, descriptor):
+        key = self.get_storage_key(descriptor)
+        return key in EphemeralUnencryptedDriver.CREDENTIALS_STORE
+
+    def load_credentials(self, descriptor):
+        key = self.get_storage_key(descriptor)
+        if key not in EphemeralUnencryptedDriver.CREDENTIALS_STORE:
+            LOGGER.error(
+                'EphemeralUnencryptedDriver failed to load_credentials: Key "%s" does not exist?',
+                key
+            )
+            return None
+
+        unencrypted_raw_creds = EphemeralUnencryptedDriver.CREDENTIALS_STORE[key]
+
+        return Credentials(unencrypted_raw_creds, False)
+
+    def save_credentials(self, descriptor, credentials):
+        """
+        Args:
+            descriptor (str):
+            credentials (Credentials):
+
+        Return:
+            bool
+        """
+        if credentials.is_encrypted():
+            unencrypted_raw_creds = credentials.get_data_kms_decrypted()
+        else:
+            unencrypted_raw_creds = credentials.data()
+
+        key = self.get_storage_key(descriptor)
+        EphemeralUnencryptedDriver.CREDENTIALS_STORE[key] = unencrypted_raw_creds
+        return True
+
+    @staticmethod
+    def clear():
+        EphemeralUnencryptedDriver.CREDENTIALS_STORE = {}
```

Comment:
You could also just call `EphemeralUnencryptedDriver.CREDENTIALS_STORE.clear()` to flush the dict instead of re-instantiating to a new one

Reply:
this may also make more sense to be a `classmethod`. referencing via the class name (`EphemeralUnencryptedDriver`) within the class isn't necessarily _wrong_, it just feels clunky if it can be avoided